### PR TITLE
(PDB-1494) Update release notes for 2.3.4.

### PR DESCRIPTION
**Pull request by wkalt**
This updates our release notes for the 2.3.4 release.

### CODE REVIEW

**Comment by rbrw** [comment]
Taking a look.

File: documentation/release_notes.markdown [hunk 1]
```diff
@@ -6,6 +6,58 @@ canonical: "/puppetdb/latest/release_notes.html"
...
+* We have removed Ubuntu 10.04 and Fedora 19 from our build targets as they are
+  end of life and no longer supported by Puppet Labs. ([PDB-1449](https://tickets.puppetlabs.com/browse/PDB-1449))
+
+* We have upgraded to the latest versions of prismatic-schema and
+  trapperkeeper-jetty9-webserver. ([PDB-1439](https://tickets.puppetlabs.com/browse/PDB-1439))
+
+#### Documentation
+
+* We have a document to describe the upgrade process from the v3 PuppetDB API
+  to v4. See [here][upgrading]. ([PDB-1421](https://tickets.puppetlabs.com/browse/PDB-1421))
```
Comment (rbrw) [inline documentation/release_notes.markdown]: @wkalt wonder about something like "We have documented the upgrade process from the v3 API to the v4 API here..." to match the tone of the other notes?

**Comment by rbrw** [comment]
@wkalt pending the few bits we discussed, looks good to me.

**Comment by wkalt** [comment]
@rbrw thanks, updated

**Comment by wkalt** [comment]
This should not be merged until it's changed to include the deprecation PRs currently in flight -- url-prefix and event counts / aggregate-event-counts.

**Pull request closed without merging**